cmd/bad-key-revoker/main.go

```diff
@@ -471,19 +477,85 @@ func main() {
 		logger:          logger,
 		clk:             clk,
 	}
+
+	// If the MaxBackoff.Duration was not set via the config, set it now
+	// so that it can't increase so high that it will not try on a
+	// reasonable timeframe.
+	if config.BadKeyRevoker.MaxBackoff.Duration == 0 {
+		config.BadKeyRevoker.MaxBackoff.Duration = time.Second * 60
+	}
+
+	// Set the backup policy
+	backoff := &BackoffPolicy{
+		Limit:   config.BadKeyRevoker.MaxBackoff.Duration,
+		Minimum: config.BadKeyRevoker.Interval.Duration,
+		Factor:  1.3,
+	}
+
+	// Run bad key revoker in a loop. Backoff if no work or errors. Increment error or backoff metrics.
 	for {
 		noWork, err := bkr.invoke()
 		if err != nil {
 			keysProcessed.WithLabelValues("error").Inc()
 			logger.AuditErrf("failed to process blockedKeys row: %s", err)
+			// Backoff on error
+			backoff.increase()
```

Comment:
Here we're increasing the backoff because there was an error, but then we're not actually backing off -- the `continue` on line 504 skips the `Sleep` on line 516.

Reply:
Good catch. I originally had the sleep at the top but moved it down to the bottom so it didn't sleep for the configured minimum (interval) on startup.

Reply:
I added a sleep at the end of the nowork or error conditions. This seems like the most reasonable approach.